### PR TITLE
Make sure event handler is called in sync order and with the lateset state.

### PR DESCRIPTION
**Pull request by tanbowensg**
# Fixed Bugs
1. Fix: event handler is called async when `wait.time` is `0`.
2. Fix: `deepEvalAndWatch` method generates mutable object.

# Changes
In order to get the latest state, every time before an event handler is going to be called, **the handler schema need to be evaled again**. Otherwise, the event handler will be called with state of last frame.

This logic needs to be added to component **manually**. But only few components need to add it because these components need to trigger events by itself, like button in `table` or `onComplete` in `fetch`. Most components' events are handled by event trait.

### CODE REVIEW

File: packages/chakra-ui-lib/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -184,13 +185,14 @@ export const TableImpl = implementTable(
...
                   {isMultiSelect ? checkbox : undefined}
-                  {columns.map(column => (
+                  {columns.map((column, j) => (
                     <TableTd
                       index={i}
                       key={column.key}
                       item={item}
-                      column={column}
                       onClickItem={() => selectItem(item)}
+                      rawColumn={(component.properties.columns as any)[j]}
```
Comment (Yuyz0112) [inline packages/chakra-ui-lib/src/components/Table/Table.tsx:194]: why do we need `any` here?

File: packages/chakra-ui-lib/src/components/Table/TableTd.tsx [hunk 2]
```diff
@@ -44,15 +52,18 @@ export const TableTd: React.FC<{
...
-            name: handler.method.name,
-            parameters: handler.method.parameters,
+            componentId: evaledHandler.componentId,
+            name: evaledHandler.method.name,
+            parameters: evaledHandler.method.parameters,
           });
         });
       };
-      content = <Button onClick={onClick}>{column.buttonConfig.text}</Button>;
+      content = <Button onClick={onClick}>{buttonConfig.text}</Button>;
```
Comment (Yuyz0112) [inline packages/chakra-ui-lib/src/components/Table/TableTd.tsx:66]: Not related to this PR: do we still need a build-in button integration when we support module renderer? Especially when the module renderer can accept any component as a module.

Comment (tanbowensg) [reply packages/chakra-ui-lib/src/components/Table/TableTd.tsx:66]: I think there could be these reasons:
1. Module is not stable now.
2. Button in table is very common and build-in button is much more convinient than a module.
3. Create a module only for a button maybe too heavy.

File: packages/chakra-ui-lib/src/components/Table/Table.tsx (hunk 1 above)
Comment (tanbowensg) [reply packages/chakra-ui-lib/src/components/Table/Table.tsx:194]: fixed.

**Review by Yuyz0112 — APPROVED** [review]

**Pull request merged**